app/models/concerns/spree/gateway/adyen_gateway.rb

```diff
@@ -21,61 +21,81 @@ def merchant_account
     end
 
     def provider_class
-      ::Adyen::API
-    end
-
-    def provider
-      ::Adyen.configuration.api_username = api_username
-      ::Adyen.configuration.api_password = api_password
-      ::Adyen.configuration.default_api_params[:merchant_account] = merchant_account
-
-      provider_class
+      ::Adyen::REST
     end
 
     def capture(amount, psp_reference, currency:, **_opts)
-      value = { currency: currency, value: amount }
+      params = payment_params(amount, currency, psp_reference)
 
       handle_response(
-        provider.capture_payment(psp_reference, value),
+        execute_request(:capture_payment, params),
         psp_reference
       )
     end
 
     def cancel(psp_reference, _gateway_options = {})
+      params = { merchant_account: merchant_account, original_reference: psp_reference }
+
       handle_response(
-        provider.cancel_or_refund_payment(psp_reference),
+        execute_request(:cancel_or_refund_payment, params),
         psp_reference
       )
     end
 
+
     def credit(amount, source = nil, psp_reference, currency: nil, **options)
       # in the case of a "refund", we don't have the full gateway_options
       currency ||= options[:originator].payment.currency
-      amount = { currency: currency, value: amount }
+      params = payment_params(amount, currency, psp_reference)
+      params.merge!(options.slice(:additional_data)) if options[:additional_data]
 
       handle_response(
-        provider.refund_payment(psp_reference, amount),
+        execute_request(:refund_payment, params),
         psp_reference
       )
     end
 
     private
 
+    def rest_client
```

Comment:
This method, and the calls like

```ruby
+        ::Adyen::REST.session(rest_client) do |client|
 +          client.list_recurring_details({
 +            merchant_account: payment_method.merchant_account,
 +            shopper_reference: reference_number_from_order,
 +          })
```
have a bit of duplication.

What do you think of moving them to a separate class to provide an abstraction over the rest client?